app/screens/address-screen/set-address-modal.tsx

```diff
@@ -0,0 +1,219 @@
+import { GaloyTextInput } from "@app/components/galoy-text-input"
+import { useI18nContext } from "@app/i18n/i18n-react"
+import { palette } from "@app/theme"
+import { useMutation } from "@galoymoney/client"
+import React, { useEffect } from "react"
+import { Modal, TouchableWithoutFeedback, View } from "react-native"
+import { Button, Text } from "react-native-elements"
+import EStyleSheet from "react-native-extended-stylesheet"
+import crashlytics from "@react-native-firebase/crashlytics"
+import { KeyboardAwareScrollView } from "react-native-keyboard-aware-scroll-view"
+import { CustomIcon } from "@app/components/custom-icon"
+import { useNavigation } from "@react-navigation/native"
+import { StackNavigationProp } from "@react-navigation/stack"
+import { RootStackParamList } from "@app/navigation/stack-param-lists"
+
+const styles = EStyleSheet.create({
+  centeredView: {
+    flex: 1,
+    justifyContent: "center",
+    alignItems: "center",
+    marginTop: 22,
+  },
+  modalView: {
+    margin: 20,
+    backgroundColor: palette.white,
+    borderRadius: 20,
+    padding: 35,
+    shadowColor: palette.midGrey,
+    shadowOffset: {
+      width: 0,
+      height: 2,
+    },
+    shadowOpacity: 0.25,
+    shadowRadius: 4,
+    elevation: 5,
+    width: "90%",
+  },
+  buttonStyle: {
+    backgroundColor: palette.primaryButtonColor,
+    color: palette.white,
+    marginTop: 32,
+  },
+  cancelTextContainer: {
+    marginTop: 20,
+    alignItems: "center",
+  },
+  cancelText: {
+    color: palette.primaryButtonColor,
+  },
+  labelStyle: {
+    marginBottom: 16,
+  },
+  errorStyle: {
+    color: palette.error,
+    marginTop: 16,
+  },
+  explainerText: {
+    fontFamily: "Roboto",
+    fontStyle: "normal",
+    fontWeight: "400",
+    fontSize: 15,
+    lineHeight: 24,
+    marginTop: 16,
+  },
+  titleText: {
+    fontFamily: "Roboto",
+    fontStyle: "normal",
+    fontWeight: "500",
+    fontSize: 18,
+    lineHeight: 24,
+    color: palette.lapisLazuli,
+  },
+  newAddressContainer: {
+    justifyContent: "center",
+    alignItems: "center",
+    backgroundColor: palette.inputBackground,
+    borderRadius: 8,
+    height: 50,
+    marginTop: 16,
+  },
+  newAddressText: {
+    fontFamily: "Roboto",
+    fontStyle: "normal",
+    fontWeight: "500",
+    fontSize: 18,
+    lineHeight: 24,
+    color: palette.lapisLazuli,
+  },
+  backText: {
+    justifyContent: "center",
+    alignItems: "center",
+    marginTop: 16,
+  },
+})
+
+type SetAddressModalProps = {
+  modalVisible: boolean
+  toggleModal?: () => void
+}
+
+export const SetAddressModal = ({ modalVisible, toggleModal }: SetAddressModalProps) => {
+  const { LL } = useI18nContext()
+  const [address, setAddress] = React.useState("")
+  const [error, setError] = React.useState("")
+  const [newAddress, setNewAddress] = React.useState("")
+  const [pristine, setPristine] = React.useState(true)
```

Comment:
```pristine``` seems to be a duplication of state that is captured by ```address```. It could be calculated each time as ```const pristine = !address``` instead of having to kept in sync with address. In order to make its purpose more explicit you could also rename it to something like ```galoyAddressHasBeenSet```.